### PR TITLE
mon: tolerate osdmap with v1 addrs and osd bound to v2+v1

### PR DESCRIPTION
**Pull request by liewegas**
Fixes: http://tracker.ceph.com/issues/38598

### CODE REVIEW

**Comment by leseb** [comment]
@guits fix for your upgrade issue?

**Comment by guits** [comment]
@leseb I couldn't test yet but yes, I think it should fix it.

**Comment by liewegas** [comment]
@neha-ojha @jecluis I added a 'mon set-addrs' command and a couple tests to verify the osd behavior.

we could definitely expand this to include other daemon types...

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -458,6 +458,11 @@ COMMAND("mon set-rank " \
 	"name=rank,type=CephInt",
 	"set the rank for the specified mon",
 	"mon", "rw")
+COMMAND("mon set-addrs " \
+	"name=name,type=CephString " \
+	"name=addrs,type=CephString",
+	"set the addrs (IPs and ports) a specific monitor binds to",
+	"mon", "rw")
```
Comment (batrick) [inline src/mon/MonCommands.h]: If this command is just for testing, shall we set the `HIDDEN` flag?

Comment (liewegas) [reply src/mon/MonCommands.h]: I think it's a useful (if low-level) command.  Without it you can't change a monitor's addr (port, ipv4 vs ipv6, etc) without removing and re-adding it.

**Pull request merged**